cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/AbstractIndexTestBase.java

```diff
@@ -29,38 +31,37 @@
 
     String factoryPath = null;
     DatastoreManager factory = null;
-    DatastoreImpl ds = null;
-    IndexManager im = null;
-    SQLDatabase db = null;
+    ForwardingDatastore fd = null;
+    QueryableDatastore ds = null;
+    SQLDatabaseQueue dbq = null;
 
     @Before
     public void setUp() throws Exception {
         factoryPath = TestUtils.createTempTestingDir(AbstractIndexTestBase.class.getName());
         assertThat(factoryPath, is(notNullValue()));
         factory = DatastoreManager.getInstance(factoryPath);
         assertThat(factory, is(notNullValue()));
-        ds = (DatastoreImpl) factory.openDatastore(AbstractIndexTestBase.class.getSimpleName());
+        ds = (QueryableDatastore) factory.openDatastore(AbstractIndexTestBase.class.getSimpleName());
         assertThat(ds, is(notNullValue()));
-        im = new IndexManager(ds);
-        assertThat(im, is(notNullValue()));
-        db = TestUtils.getDatabaseConnectionToExistingDb(im.getDatabase());
-        assertThat(db, is(notNullValue()));
-        assertThat(im.getQueue(), is(notNullValue()));
-        String[] metadataTableList = new String[] { IndexManager.INDEX_METADATA_TABLE_NAME };
-        SQLDatabaseTestUtils.assertTablesExist(TestUtils.getDatabaseConnectionToExistingDb(db),
-                                               metadataTableList);
+        fd = new ForwardingDatastore(ds);
+        dbq = ds.getQueryQueue();
+        assertThat(fd, is(notNullValue()));
+//        db = TestUtils.getDatabaseConnectionToExistingDb(im.getDatabase());
+//        assertThat(db, is(notNullValue()));
+//        assertThat(im.getQueue(), is(notNullValue()));
+        String[] metadataTableList = new String[] { QueryConstants.INDEX_METADATA_TABLE_NAME };
+//        SQLDatabaseTestUtils.assertTablesExist(TestUtils.getDatabaseConnectionToExistingDb(db),
+//                                               metadataTableList);
     }
 
     @After
     public void tearDown() {
-        im.close();
-        assertThat(im.getQueue().isShutdown(), is(true));
-        TestUtils.deleteDatabaseQuietly(db);
         ds.close();
+//        assertThat(im.getQueue().isShutdown(), is(true));
+//        TestUtils.deleteDatabaseQuietly(db); // FIXME make this work
```

Comment:
Not sure, since the directory is deleted, it'll probably make more sense to just make sure it's closed and then delete.